### ISSUE TITLE
GPTScore: A Comprehensive Framework for Evaluating Text Generation

### ISSUE THREAD

**Issue by ShellLM** [irthomasthomas/undecidability#819]
- [ ] [GPTScore: A Comprehensive Framework for Evaluating Text Generation](https://paperswithcode.com/paper/gptscore-evaluate-as-you-desire)

# GPTScore: A Comprehensive Framework for Evaluating Text Generation

## Snippet
Here is the text with the duplicated sections removed:

### Notes for GPTScore: Evaluate as You Desire

| Aspect | Task | Definition |
|--------|------|------------|
| Semantic Coverage (COV) | Summ | How many semantic content units from the reference text are covered |
| Factuality (FAC) | Summ | Does the generated text preserve the factual statements of the source |
| Consistency (CON) | Summ, Diag | Is the generated text consistent in the information it provides? |
| Informativeness (INF) | Summ, D2T, Diag | How well does the generated text capture the key ideas of its source |
| Coherence (COH) | Summ, Diag | How much does the generated text make sense? |
| Relevance (REL) | Diag, Summ, D2T | How well is the generated text relevant to its source text? |
| Fluency (FLU) | Diag, Summ, D2T, MT | Is the generated text well-written and grammatical? |
| Accuracy (ACC) | MT | Are there inaccuracies, missing, or unfactual content in the generated text? |
| Multidimensional Quality Metrics (MQM) | MT | How is the overall quality of the generated text? |
| Interest (INT) | Diag | Is the generated text interesting? |
| Engagement (ENG) | Diag | Is the generated text engaging? |
| Specific (SPE) | Diag | Is the generated text generic or specific to the source text? |
| Correctness (COR) | Diag | Is the generated text correct or was there a misunderstanding of the source? |
| Semantically appropriate (SEM) | Diag | Is the generated text semantically appropriate? |
| Understandability (UND) | Diag | Is the generated text understandable? |
| Error Recovery (ERR) | Diag | Is the system able to recover from errors that it makes? |
| Diversity (DIV) | Diag | Is there diversity in the system responses? |
| Depth (DEP) | Diag | Does the system discuss topics in depth? |
| Likeability (LIK) | Diag | Does the system display a likeable personality? |
| Flexibility (FLE) | Diag | Is the system flexible and adaptable to the user and their interests? |
| Inquisitiveness (INQ) | Diag | Is the system inquisitive throughout the conversation? |

### Emergent Ability
Recent works progressively reveal a variety of emergent abilities of generative pre-trained language models with appropriate tuning or prompting methods, such as in-context learning (Min et al., 2022), chain-of-thought reasoning (Wei et al., 2022), and zero-shot instruction (Ouyang et al., 2022). One core commonality of these abilities is to allow for handling customized requirements with a few or even zero annotated examples. It's the appearance of these abilities that allows us to re-invent a new way for text evaluation–evaluating from the textual description, which can achieve customizable, multi-faceted, and train-free evaluation.

The core idea of GPTS CORE is that a generative pre-training model will assign a higher probability of high-quality generated text following a given instruction and context. In our method, the instruction is composed of the task description d and the aspect definition a.

### Few-shot with Demonstration
The generative pre-trained language model can better perform tasks when prefixed with a few annotated samples (i.e., demonstrations). Our proposed framework is flexible in supporting this by extending the prompt template T with demonstrations. Choice of Prompt Template Prompt templates define how task description, aspect definition, and context are organized. Mingling desirable prompts itself is a non-trivial task and there are extensive research works there (Liu et al., 2021; Fu et al., 2022). In this work, for the GPT3-based model, we opt for prompts that are officially provided by OpenAI. [OpenAI Examples](https://beta.openai.com/examples)

#### Tasks, Datasets, and Aspects
To achieve a comprehensive evaluation, in this paper, we cover a broad range of natural language generation tasks: Dialogue Response Generation, Text Summarization, Data-to-Text, and Machine Translation, which involves 37 datasets and 22 evaluation aspects in total.

1. **Dialogue Response Generation** aims to automatically generate an engaging and informative response based on the dialogue history. Here, we choose to use the FED datasets and consider both turn-level and dialogue-level evaluations.

2. **Text Summarization** is a task of automatically generating informative and fluent summary for a given long text. Here, we consider the following four datasets, SummEval, REALSumm, NEWSROOM, and QAGS_XSUM, covering 10 aspects.

3. **Data-to-Text** aims to automatically generate a fluent and factual description for a given table. Our work considered BAGEL and SFRES datasets.

4. **Machine Translation** aims to translate a sentence from one language to another. We consider a subdatasets of Multidimensional Quality Metrics (MQM), namely, MQM-2020 (Chinese->English).

#### Metric Comparison
Tab. 7 summarizes several popular generated text evaluation methods.

### Unsupervised Evaluation of Interactive Dialog with DialoGPT
Shikib Mehri, Maxine Eskenazi

#### Abstract
It is important to define meaningful and interpretable automatic evaluation metrics for open-domain dialog research. Standard language generation metrics have been shown to be ineffective for dialog. This paper introduces the FED metric (fine-grained evaluation of dialog), an automatic evaluation metric which uses DialoGPT, without any fine-tuning or supervision. It also introduces the FED dataset which is constructed by annotating a set of human-system and human-human conversations with eighteen fine-grained dialog qualities. The FED metric (1) does not rely on a ground-truth response, (2) does not require training data and (3) measures fine-grained dialog qualities at both the turn and whole dialog levels. FED attains moderate to strong correlation with human judgement at both levels.

### Main Observations
GPTScore: Evaluate(1) The introduction of instruction (IST) significantly improves the performance in three different aspects of ACC, FLU, and MQM. In Tab. 4, the average performance of 19 GPTS CORE based evaluators with instruction (IST) significantly outperforms vanilla (VAL). (2) The combination of instruction and demonstration (IDM) brings gains for the evaluator with different model structures. In Tab. 4, the performance of GPT3, GPT2, OPT, and FT5 improves a lot when instruction and demonstration (IDM) are introduced. (3) The evaluator built based on GPT3-c01 achieves comparable performance with GPT3-d01 and GPT3-d03. Since the GPT3-d01 and GPT3-d03 are most expensive variant of GPT3, the cheaper and comparative GPT3-c01 is a good choice for machine translation task.

### Conclusion
In this paper, we propose to leverage the emergent abilities from generative pre-training models to address intricate and ever-changing evaluation requirements. The proposed framework, GPTSCORE, is studied on multiple pre-trained language models with different structures, including the GPT3 with a model size of 175B. GPTS CORE has multiple benefits: customizability, multi-faceted evaluation, and train-free, which enable us to flexibly craft a metric that can support 22 evaluation aspects on 37 datasets without any learning process yet attain competitive performance. This work opens a new way to audit generative AI by utilizing generative AI.

**URL:** [GPTScore: Evaluate as You Desire](https://paperswithcode.com/paper/gptscore-evaluate-as-you-desire)

#### None

**Comment by ShellLM** [comment]
### Related content
#811 similarity score: 0.92
#706 similarity score: 0.91
#817 similarity score: 0.91
#715 similarity score: 0.89
#762 similarity score: 0.89
#684 similarity score: 0.88
